changelogs/unreleased/5752-davinci26-major.md

```diff
@@ -0,0 +1,14 @@
+## Fix bug with algorithm used to sort Envoy regex/prefix path rules
+
+Envoy is greedy matching path routes and as a result order is important. Contour to produce consistent routing tables in the presence of HTTPProxy inclusion and Ingress Conformance sorts the routes before updating Envoy.
```

Comment:
```suggestion
Envoy greedy matches routes and as a result the order route matches are presented to Envoy is important. Contour attempts to produce consistent routing tables so that the most specific route matches are given preference. This is done to facilitate consistency when using HTTPProxy inclusion and provide a uniform user experience for route matching to be inline with Ingress and Gateway API Conformance.
```

Reply:
fixed in de92073f94f2426279a4f99d39029d97355b61d2